src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -887,6 +920,57 @@ public void testGetImportedKeysDW() throws SQLException {
         }
     }
 
+    @Test
+    public void testValidateColumnMetadata() throws SQLException {
```

Comment:
I don't think this test actually test the fix, it just tests that all the columns are there and in the order expected, which would never fail because that's the order we put them in. What the issue that the classes weren't cast to the correct types so they're not assignable. Take a look at the repro in the origin issue it shows what the problem is and what we should be checking for

Reply:
Logic has been modified to check the JDBC class types in the metadata result set, instead of checking the SQL Server data types for the result set. The logic was greatly simplified by using in Integer class instead of a generic Object class as the expected class for the BUFFER_LENGTH column (i.e. which is not actually used in the JDBC spec).